pkg/resource/template/util.go

```diff
@@ -17,10 +20,11 @@ type fileInfo struct {
 	Md5  string
 }
 
-func appendPrefix(prefix string, keys []string) []string {
+func expandKeys(prefix string, keys []string) []string {
 	s := make([]string, len(keys))
 	for i, k := range keys {
-		s[i] = path.Join(prefix, k)
+		// Prepend the prefix and replace "NODENAME" in the key by the actual node name.
+		s[i] = path.Join(prefix, strings.Replace(k, "NODENAME", NodeName, 1))
```

Comment:
Will this substitution work? I think it's assuming we never use NODENAME in any name. Wouldn't, for example, a bgppeer named NODENAME might break this?

Reply:
I believe it's watertight as is.  The substitution is only performed on the "prefix" and "keys" settings from confd TOML config files, and I don't believe there's any reasonable use case for those to name a specific resource whose path includes NODENAME.

That said, I could add an extra layer of defence by:
- changing the uses from (for example) "/bgp/v1/host/NODENAME" to "/bgp/v1/host//NODENAME" - i.e. doubling the slash before "NODENAME"
- `strings.Replace(k, "//NODENAME", "/"+NodeName, 1)`.

WDYT?

Reply:
(I quite like the extra layer idea, so I've provisionally added a commit here to do that.)